src/components/patch/PatchContainer.tsx

```diff
@@ -230,6 +230,11 @@ export class PatchContainer extends React.Component<Props, State> {
           username = urlParams.get("user");
         }
       }
+      const pathname = window.location.hash.split("?")[0].split('/')
+      let projectName = ''
+      if(pathname.length === 4 && pathname[2] === 'project') {
+        projectName = pathname[3]
+      }
```

Comment:
might be better off using a library like [query-string](https://www.npmjs.com/package/query-string). Otherwise this function should be a util and unit tested. my vote is for the query-string approach.